include/boost/gil/detail/math.hpp

```diff
@@ -28,6 +48,158 @@ inline detail::kernel_2d<T, Allocator> get_identity_kernel()
     return kernel;
 }
 
+// Please refer https://stackoverflow.com/a/10032882/14958679 for getting an overview of the
+// concept applied for obtaining higher order Sobel kernels.
+
+/// \defgroup KernelGeneration
+/// \brief Contains documentation for functions used in kernel generation.
+///
+/// Separate functions are used for generating only those kernels whose dimensions are greater than
+/// 5x5. Smaller kernels are fed directly to the algorithm.
+///
+
+/// \addtogroup KernelGeneration
+/// @{
+
+/// \brief Produces higher order kernel vector by performing discrete convolution between lower
+/// order kernel vector and a smoothing kernel vector(kernel used for suppressing noise).
+/// \param kernel1 - First argument for kernel vector convolution.
+/// \param kernel2 - Second argument for kernel vector convolution.
+/// \tparam T1 - Type of first argument for kernel vector convolution.
+/// \tparam T2 - Type of second argument for kernel vector convolution.
+template<typename T1, typename T2>
+static auto kernel_convolve_impl(T1 kernel1, T2 kernel2) -> std::vector<std::vector<float>>
```

Comment:
It's a free function, in a header, in `detail` namespace, what is the `static` for?

Reply:
I was not using ```static``` initially which caused all builds to fail. After an exhaustive search of raw logs, I came across the following line ,  
```
2021-03-04T04:26:54.6586533Z sample_image.cpp:(.text+0x0): multiple definition of `boost::gil::detail::kernel_vector_fill(std::vector<std::vector<float, std::allocator<float> >, std::allocator<std::vector<float, std::allocator<float> > > >&, boost::gil::detail::kernel_type)'
```
This lead me to believe that the compiler was considering different definitions of ```kernel_vector_fill``` whenever it was parsing through it. For tackling this, the immediate solution which came to my mind was using ```static```. I wonder if that was the best way to solve it. 
[Here ](2021-03-04T04:26:54.6586533Z sample_image.cpp:(.text+0x0): multiple definition of `boost::gil::detail::kernel_vector_fill(std::vector<std::vector<float, std::allocator<float> >, std::allocator<std::vector<float, std::allocator<float> > > >&, boost::gil::detail::kernel_type)')is the link to ```raw log files``` for your reference.
Note : I didn't encounter this problem on my system. I used ```cmake``` on a  ```Ubuntu 20.04 OS```.

Reply:
@mloskot, is this resolvable?

Reply:
You should be `inline` instead, I think.

Reply:
I googled this a bit and I think I now understand your reason. I am attaching these links to [ODR](https://en.m.wikipedia.org/wiki/One_Definition_Rule#:~:text=The%20One%20Definition%20Rule%20) and [namespace functions](https://stackoverflow.com/a/10493005/14958679) here just in case if someone questions why I made it inline despite the function being large in size and its execution time is not lesser than its call time.

Reply:
Yes, I think this sums it up well:
["Only short functions should be inlined" is a rule of thumb, but there are so many exceptions and so many other more important reasons why you should or should not inline a function so as to make the rule of thumb almost useless.](https://stackoverflow.com/questions/3973218/header-only-libraries-and-multiple-definition-errors#comment4250148_3973236)

Reply:
No objections from my side.

Reply:
Please update the PR then